qcodes/instrument_drivers/Lakeshore/Model_325.py

```diff
@@ -278,41 +269,23 @@ def __init__(self, parent: 'Model_325', name: str, inp: str) -> None:
             vals=Numbers(min_value=1, max_value=35)
         )
 
-    def decode_sensor_status(self, sum_of_codes: int) -> str:
-        """
-        The sensor status is one of the status codes, or a sum thereof. Multiple
-        status are possible as they are not necessarily mutually exclusive.
-
-        args:
-            sum_of_codes (int)
-        """
-        components = list(self.sensor_status_codes.keys())
-        codes = self._get_sum_terms(components, int(sum_of_codes))
-        return ", ".join([self.sensor_status_codes[k] for k in codes])
-
     @staticmethod
-    def _get_sum_terms(components: list, number: int):
-        """
-        Example:
-        >>> components = [0, 1, 16, 32, 64, 128]
-        >>> _get_sum_terms(components, 96)
-        >>> ...[64, 32]  # This is correct because 96=64+32
-        """
-        if number in components:
-            terms = [number]
-        else:
-            terms = []
-            comp = np.sort(components)[::-1]
-            comp = comp[comp <= number]
-
-            while len(comp):
-                c = comp[0]
-                number -= c
-                terms.append(c)
-
-                comp = comp[comp <= number]
-
-        return terms
+    def decode_sensor_status(sum_of_codes: int) -> str:
+        class Status(IntFlag):
+            sensor_units_overrang = 128
+            sensor_units_zero = 64
+            temp_overrange = 32
+            temp_underrange = 16
+            invalid_reading = 1
+
+        if sum_of_codes == 0:
+            return 'OK'
+        sensor_status = []
+        for st in Status:
+            if st <= sum_of_codes:
```

Comment:
this `if` stuff should not be needed because the `sum_of_codes` can be converted to a `Status` instance, and then checked against containing separate flags in it, like this (the check for `0` is still needed unfortunately):
```python
total_status = Status(sum_of_codes)
if total_status == 0:
    return 'OK'
status_messages = [st.name.replace('_', ' ') for st in Status if st in total_status]
return ', '.join(status_messages)
```

Moreover, I read that although `IntFlag` exists, preference should be given to use of `Flag` and the reason for this is that `Flag` is less forgiving to `sum_of_codes` - for example, `Flag` will raise an exception on `total_status = Status(sum_of_codes)` if `sum_of_codes` contains codes that are not in the enumeration, for example, if `sum_of_codes == 16+2`.

Reply:
This is really interesting, I never knew about the Flags. You can do stuff like : 

```python
from enum import IntFlag

class Status(IntFlag):
    sensor_units_overrang = 128
    sensor_units_zero = 64
    temp_overrange = 32
    temp_underrange = 16
    invalid_reading = 1

total_status = Status(128)
Status.sensor_units_overrang in Status(128)  # Returns True 
Status.sensor_units_zero in Status(128)  # Returns False 
```

Reply:
Wow, very nice use of `IntFlag`, I changed the `decode_sensor_status` method accordingly.